### DISCUSSION TITLE
Yoshan roaster and artisan problem

### DISCUSSION THREAD

**Discussion by Kalebgriffin** [artisan-roaster-scope/artisan#780] [Q&A] [answered]
Hello. I just got my yoshan dy-1 roaster And have been setting it up. I I am hooking my roaster up to my HP windows 10 laptop. The roaster did not come with a manual or a manual on how to set up artisan. I went to the artist and website and download it drivers to get started. Although artisans did not list a specific driver that the Yoshan roaster uses. When I hook the roaster up to my laptop, and start artisan everything seems to be normal. My temperature readings are showing up on my laptop as they are on the roaster which is normal. I go to start the roast and everything is still normal, but my problem occurs when I hit the ignite button for the gas on my roaster. When I hit the ignite button the temperature read outs on my arson profile on my laptop disappear and show “u.u” In the temperature read out spots. It also says “modbus communication error” in the top left-hand corner. I will reset artisan and the temperatures will show back up as normal until I hit the ignite button again and then the problem occurs all over. Im assuming my driver is correct or it wouldn’t even show the temps in the beginning? Has this ever happened to anyone? Or does anyone have any idea on how to fix this?

**Comment by poundy** [comment]
There's no attached video - upload it to YT or somewhere and post a link.

But this is pretty easy to guess what's going on - it's electrical interference. When your ignition circuit starts up, it's using pulses of high energy to create a spark. This is in turn affecting the circuitry that's allowing Artisan to communicate to the roaster. Note: this is affecting the circuitry, so all products, not just Artisan, would have the same problem.

resolving this is simple - well not really. This problem is best solved by the manufacturer doing proper electrical isolation and testing. Anything less is just a work around, and the problem may still be seen at some point in later use.

My first test would be to see if there's a difference when your laptop is powered or running purely from batteries - is there a ground reference issue. If it runs OK when on battery alone, do that.  If not, the next change I would try to make is to use a high quality shielded USB cable, preferably one with a ferrite bead attached, to replace the one you currently have?

**Reply by Kalebgriffin** [reply]
Here is the video https://youtube.com/shorts/ugs9QXwGAUA?feature=share

at first I didn’t know what it was, but it makes sense that there is an interference somewhere due to it only causing a modbus error when I hit ignition. I’m not very savvy when it comes to all this stuff which Is why I’m posting here. I have been using my laptop on battery only without power supply, and I have not tried with the laptop on its power supply yet. Someone on Facebook had mentioned the ferrite beads so I plan on getting them to see if it helps. I have also emailed yoshan, the manufacturer about the problem but have not heard back yet. Do you have a specific cable that you would recommend?

**Reply by poundy** [reply]
no, i have no suggestion on possible cables. It might not help, but it can't hurt. I couldn't really recommend one anyhow as I don't know what kind of port you have on your PC and what is on the roaster - not all machines are the same. You could also try a generic ferrite bead/choke that clips on over the existing cable you have, and see if that helps.

Given you're already using the laptop without a power pack, it can't hurt to try it the other way around, and run it with a power supply (most issues that relate to grounded thermocouples and noisy signals tend to be fixed when removing the power supply - so personally I think this combination you're already running in has the best chance of working).

Are you able to see any additional information from the Artisan log and messages? Get them from the Help menu.  You might also try turning on serial logging and see if that shows up any more details.

**Comment by fcebreros** [comment]
I agree with pundy, there is no easy way to fix this if the ignition is
spark
based.
I can only recommend turning on the heat before connecting your laptop to
the roaster.
If not already factory done, ground the body of the roaster.
Use ungrounded thermocouples if they are easy to swap and if you have
information about the thermocouple type.
If possible, change the ignition system.
Spark systems use 5000-10000 volts sparks for ignition, it's very hard to
avoid the electrical noise they generate.

FC

**Reply by Kalebgriffin** [reply]
That is probably how I will have to end up doing it. I purchased a new USB cable with ferrite beads just to see if it works. If not then I’ll just have to keep it turned on and a very low kpa soak

**Reply by bigl00z3** [reply]
I wonder if a USB isolator could help:

[https://www.phidgets.com/?tier=3&catid=58&pcid=51&prodid=196]()
![image](https://user-images.githubusercontent.com/8886922/148031902-c46691d5-af6b-4255-a592-eef408e43682.jpeg)

But the problem is that you will only know if it helps once you have purchased it and tested it. The isolator is 58$ and the acrylic enclosure is 8$.

It has the advantage of shielding your computer from the roaster which may increase the life of your computer if you think it gets unwanted voltage from the roaster.

Nico

**Comment by CarefreeBuzzBuzz** [comment]
This is really the missing piece and I guess with habits it just wasn’t staring me in the face.  Yes turn on the burners before you start Artisan and then don’t them off.  You can reduce them to a minimal setting if you need to after you start.  



----------------------------------------
Michael Herbert, Head Roaster
Order at www.carefreebuzzbuzz.com <http://www.carefreebuzzbuzz.com/>
Artisan RoasterScope  <https://artisan-scope.org/>Team

**Comment by fcebreros** [comment]
I have tried this kind of cheap isolator's (several models) and had no luck.
For one part they don't work with some devices and most of the time they
"burned/broke" when I activated the spark.
Best
FC

**Reply by bigl00z3** [reply]
I never had issues with the ones from [https://www.phidgets.com/](url) . I have one in between my Modbus interface and the computer). 
Most Phidgets seem to be quite well designed. 

But if they burn with the spark I would be scared for the computer...

Nico

**Comment by fcebreros** [comment, marked as answer]
Yes! The voltage that can be induced/leaked into the cables can be quite
high, that's why I recommend connecting
the usb after turning on the fire.

FC
